_search-plugins/knn/performance-tuning.md

```diff
@@ -32,31 +32,86 @@ Take the following steps to improve indexing performance, especially when you pl
-
-* **Increase the number of indexing threads**
+### Disable replicas (no OpenSearch replica shard)
 
-   If the hardware you choose has multiple cores, you can allow multiple threads in native library index construction by speeding up the indexing process. Determine the number of threads to allot with the [knn.algo_param.index_thread_qty]({{site.url}}{{site.baseurl}}/search-plugins/knn/settings#cluster-settings) setting.
+   Set replicas to `0` to prevent duplicate construction of native library indexes in both primary and replica shards. When you enable replicas after indexing finishes, the serialized native library indexes are directly copied. If you have no replicas, losing nodes might cause data loss, so it's important that the data lives elsewhere so this initial load can be retried in case of an issue.
 
-  Keep an eye on CPU utilization and choose the correct number of threads. Because native library index construction is costly, having multiple threads can cause additional CPU load.
+### Increase the number of indexing threads
+
+If the hardware you choose has multiple cores, you can allow multiple threads in native library index construction by speeding up the indexing process. Determine the number of threads to allot with the [knn.algo_param.index_thread_qty]({{site.url}}{{site.baseurl}}/search-plugins/knn/settings#cluster-settings) setting.
+
+Keep an eye on CPU utilization and choose the correct number of threads. Because native library index construction is costly, having multiple threads can cause additional CPU load.
+
+
+### (Expert-level) Disable storing of vector fields in source
+
+The `_source` field contains the original JSON document body that was passed at index time. This field is not indexed and is not searchable but is stored so that it can be returned when executing fethc requests such as `get` and `search`. When using vector fields within the source, you can remove the vector field to save up disk space, as shown in the following example where the `location` vector is excluded:
+
+  ```json
+  PUT /<index_name>/_mappings
+  {
+      "_source": {
+        "excludes": ["location"]
+      },
+      "properties": {
+          "location": {
+              "type": "knn_vector",
+              "dimension": 2,
+              "method": {
+                  "name": "hnsw",
+                  "space_type": "l2",
+                  "engine": "faiss"
+              }
+          }
+      }
+  }
+  ```
+
+
+Disabling the `_source` field can cause certain featues to become not supported, such as `update`, `update_by_query`, and `reindex` APIs and the ability to debug queries or aggregations by using the original document at index time.
+
+In OpenSearch 2.15 or great, you can further improve the indexing speed and reduce disk space, by removing remove vector field from `_recovery_source`, as shown in the following example:
```

Comment:
Should "vector" be in code font?

Reply:
No, since its referring to vector fields generally.